.circleci/config.yml

```diff
@@ -69,17 +91,29 @@ jobs:
             name: Download dependencies
             command: ./gradlew androidDependencies
         - *save_cache
+#       It's faster if we build before running the quality checks, because quality checks depend on build output anyway.
+#       Also, while running gradle on daemon mode, we can even run build targets individually without impacting runtime ramp-up time.
+#       This also implies that we can also observe per-target build timings, instead of just the total execution time from a single command.
+        - run:
+            name: Assemble debug build
+            command: |
+                ./gradlew assembleDebug -PdisablePreDex
+        - run:
+            name: Assemble test build
+            command: |
+                ./gradlew assembleDebugAndroidTest -PdisablePreDex
         - run:
             name: Run code quality checks
-            command: ./gradlew pmd checkstyle lint findbugs
+#           When running the plain lint target, it will execute by default for all flavors.
+#           This implies building the release and test flavors as well, which are out of scope for this job.
+#           Instead, using lintDebug it's running only once, for the 'debug' flavor that we are building here
+#           This means ~3x faster for the lint task.
+            command: ./gradlew pmd checkstyle lintDebug findbugs
```

Comment:
It feels like 3x speed up is a worthy tradeoff, but, I have to ask. How likely is it that there will be an issue findable by lint that is in the release build that is not in the debug build?

Reply:
Good point. Indeed we could miss issues on what was not analyzed/checked. 
But actually, the flavor difference itself is just a fraction of the project (in fact it's just the build configuration).
On the other hand, the debug build shouldn't be concerned with other configurations and the way it used to be before this change, every debug was burdened (although out of scope).

There are two approaches I'd weight-balance here:
1. run lintRelease or lintOdkCollectRelease on every debug build (and drop lintDebug entirely from the debug build)
2. run lintDebug on debug builds and run lintRelease or lintOdkCollectRelease on release builds

With (1) during debug build, we'd know any release-flavor lint violations early, but we may miss debug version violations. I don't like this approach because it's a compromise.
With (2) the debug build deals with it's own quality and the release build is separately concerned with it's own quality

My suggestions is to go with #2 and keep the concern within it's scope. I'd rather re-run the quality checks on release build to double-check them, rather than run extra release checks on every debug build.

Reply:
Agreed with this approach! I'll note this in our release process.